torchbenchmark/score/compute_score.py

```diff
@@ -8,31 +8,147 @@
 import sys
 import os
 import yaml
+import importlib
 
 from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
 
 SPEC_FILE_DEFAULT = "score.yml"
 TARGET_SCORE_DEFAULT = 1000
 
-def compute_score(config, data, fake_data=None):
-    target = config['target']
-    score = 0.0
+def get_benchmark_norms(data):
+    """
+    Helper function which gets the normalization values per benchmark
+    by going through the reference data file.
+
+    This reference data can be user provided or by default the first file
+    in the directory of data files is considered as the reference data file
+    """
+    benchmark_norms = {b['name']: b['stats']['mean'] for b in data['benchmarks']}
+    return benchmark_norms
+
+def get_weights_from_spec(spec):
+    """
+    Calculates the static benchmark weights by iterating the spec
+    file and constructs a b_weight dictionary with (key, value) pair
+    is (task, weight_for_benchmark_per_task)
+
+    The spec here can be user defined .yaml file or if no spec is defined,
+    the default score.yaml is used
+    """
+    b_weight = defaultdict(float)
+    category_spec = spec['hierarchy']['model']
+    domain_weight = 1.0/ len(category_spec)
+    for domain in category_spec:
+        tasks = category_spec[domain]
+        task_weight = 1.0 / len(tasks)
+        for task in tasks:
+            benchmarks = tasks[task]
+            benchmark_weight = 1.0 / len(benchmarks)
+            b_weight[task] = domain_weight * task_weight * benchmark_weight
+
+    return b_weight
+
+
+def get_benchmark_configuration(b):
+    """
+    This helper function extracts the configuration details
+    from the benchmark name.
+    For eg., if benchmark_name = test_eval[pytorch_struct-cuda-jit]
+            test, device, mode = eval, cuda, jit
+            model_name = pytorch_struct
+
+    This helper also extracts the task and model_name of the benchmark
+    The function returns a tuple of the configs, model task, model_name
+    """
+    test, device, mode = "train", "cuda", "eager"
+    if "eval" in b:
+        test = "eval"
+    if "cpu" in b:
+        device = "cpu"
+    if "jit" in b:
+        mode = "jit"
+
+    # Extract the model name from the benchmark b
+    pos1 = b.index("[")
+    pos2 = b.index("-"+device)
+    model_name = b[pos1+1:pos2]
+
+    # Get the Model task value by reading the Model attributes.
+    p = Path(__file__).parent.joinpath('../../models/'+ model_name)
```

Comment:
do we record the task into the benchmark json? if not we probably should, so we can just read it form the data instead of having to reconstruct it this way.  

also, if we are going to do this, it'd probably be better to make this lookup a helper function so we could possibly improve how it's implemented later.  something like get_model_task(model_name)